core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -741,18 +742,18 @@ class DAGScheduler(
       val stageInfos = stageIds.flatMap(id => stageIdToStage.get(id).map(_.latestInfo))
       listenerBus.post(
         SparkListenerJobStart(job.jobId, jobSubmissionTime, stageInfos, properties))
-      submitStage(finalStage)
+      submitStage(finalStage, Some(missingStages))
     }
     submitWaitingStages()
   }
 
   /** Submits stage, but first recursively submits any missing parents. */
-  private def submitStage(stage: Stage) {
+  private def submitStage(stage: Stage, missingStages: Option[List[Stage]] = None) {
```

Comment:
+1, it actually simplifies the code down there because now you don't need the `getOrElse` in L756 anymore